app/src/main/java/org/sil/storyproducer/model/BLBookList.kt

```diff
@@ -1,40 +1,294 @@
 package org.sil.storyproducer.model
 
+import android.os.AsyncTask
+import android.util.Xml
+import org.sil.storyproducer.App
+import org.sil.storyproducer.R
+import org.sil.storyproducer.controller.bldownload.BLDownloadActivity
+import org.xmlpull.v1.XmlPullParser
+import org.xmlpull.v1.XmlPullParserException
+import java.io.IOException
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.net.URL
 import java.util.*
 
+class BLBook(
+    val Title: String,
+    val LangCode: String,
+    val ThumbnailURL: String,
+    val BloomSourceURL: String)
+
+object BlBookList : BLBookList(Date())
+
 /**
  * BLBookList is used as a collection of books available to download from BloomLibrary
  */
 open class BLBookList(var dateUpdated: Date) {
-    var booklist = mutableListOf<BLBook>()
+
+    companion object {
+
+        var booklistLoading = false
+        var booklist : List<BLBook>? = null
+
+        const val WIFI = "Wi-Fi"
+        const val ANY = "Any"
+        const val BLOOM_LIB_URL = "https://sil-storyproducer-resources.s3.amazonaws.com/bl1/bl_catalog.xml"
```

Comment:
Hi Peter, could we please rename the file as bl_samples.xml? I now think that's going to be clearer in the public code. Or I can change this after it's merged into sillsdev.